src/Data/Text/Encoding/Base64.hs

```diff
@@ -39,9 +43,33 @@ encodeBase64 = B64.encodeBase64 . T.encodeUtf8
 -- See: <https://tools.ietf.org/html/rfc4648#section-4 RFC-4648 section 4>
 --
 decodeBase64 :: Text -> Either Text Text
-decodeBase64 = fmap T.decodeUtf8 . B64.decodeBase64 . T.encodeUtf8
+decodeBase64 = fmap T.decodeLatin1 . B64.decodeBase64 . T.encodeUtf8
```

Comment:
Why is this decoded as Latin1? Won't this create funky text if it was UTF8 originally? (e.g. Chinese / Japanese)

Reply:
Garbage in, garbage out. This will always return the correct decoded string given a correctly encoded input value. What it won't do, in contrast to `decodeUtf8`, is throw an exception. `decodeLatin1` is the _only_ total conversion function from ASCII to text.

Reply:
So yes, if you use this, you accept the caveat that you must validate your strings before input, or use `decodeWith` if you require specific charset validation after decode.

Reply:
Ah, I didn't notice you added `decodeWith`, nice!